src/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -0,0 +1,66 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.Identity.Client.ApiConfig
+{
+    /// <summary>
+    /// </summary>
+    public sealed class AcquireTokenForClientParameterBuilder :
+        AbstractAcquireTokenParameterBuilder<AcquireTokenForClientParameterBuilder, IAcquireTokenForClientParameters>
+    {
+        /// <summary>
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns></returns>
+        internal static AcquireTokenForClientParameterBuilder Create(IEnumerable<string> scopes)
+        {
+            return new AcquireTokenForClientParameterBuilder().WithScopes(scopes);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="forceRefresh"></param>
+        /// <returns></returns>
+        public AcquireTokenForClientParameterBuilder WithForceRefresh(bool forceRefresh)
+        {
+            Parameters.ForceRefresh = forceRefresh;
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="withCertificate"></param>
+        /// <returns></returns>
+        public AcquireTokenForClientParameterBuilder WithCertificate(bool withCertificate)
```

Comment:
I don't understand this override? would we want to override the certificate set in the CCA builder?

Reply:
In our current public API, we have:

AcquireTokenForClientAsync(IEnumerable<string> scopes)
AcquireTokenForClientAsync(IEnumerable<string> scopes, bool forceRefresh)
AcquireTokenForClientWithCertificateAsync(IEnumerable<string> scopes)
AcquireTokenForClientWithCertificateAsync(IEnumerable<string> scopes, bool forceRefresh)

The WithCertificate() allows us to differentiate between the AcquireTokenForClient and AcquireTokenForClientWithCertificate versions above.  I'm open to suggestions on how to improve this API surface.  This was put here to ensure we can cover all existing permutations of the public API.


---
In reply to: [246268229](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/766#discussion_r246268229) [](ancestors = 246268229)

Reply:
Updated this to be WithSendX5C as we discussed today.

---
In reply to: [246501111](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/766#discussion_r246501111) [](ancestors = 246501111,246268229)